tests/components/mqtt/test_light_json.py

```diff
@@ -642,203 +371,9 @@ async def test_no_color_brightness_color_temp_if_no_topics(
-    assert light_state.attributes.get("color_temp_kelvin") == 6451  # 155 mired
-
-    async_fire_mqtt_message(hass, "test_light_rgb", '{"state":"ON", "color_temp":null}')
-
-    light_state = hass.states.get("light.test")
-    assert light_state.attributes.get("color_temp_kelvin") is None
-
-    async_fire_mqtt_message(
-        hass, "test_light_rgb", '{"state":"ON", "effect":"colorloop"}'
-    )
-
-    light_state = hass.states.get("light.test")
-    assert light_state.attributes.get("effect") == "colorloop"
-
-    async_fire_mqtt_message(
-        hass,
-        "test_light_rgb",
-        '{"state":"ON",'
-        '"color":{"r":255,"g":255,"b":255},'
-        '"brightness":128,'
-        '"color_temp":155,'
-        '"effect":"colorloop"}',
-    )
-    light_state = hass.states.get("light.test")
-    assert light_state.state == STATE_ON
-    assert light_state.attributes.get("brightness") == 128
-
-    async_fire_mqtt_message(
-        hass,
-        "test_light_rgb",
-        '{"state":"OFF","brightness":0}',
-    )
-    light_state = hass.states.get("light.test")
-    assert light_state.state == STATE_OFF
-    assert light_state.attributes.get("brightness") is None
-
-    # Simulate the lights color temp has been changed
-    # while it was switched off
-    async_fire_mqtt_message(
-        hass,
-        "test_light_rgb",
-        '{"state":"OFF","color_temp":201}',
-    )
-    light_state = hass.states.get("light.test")
-    assert light_state.state == STATE_OFF
-    # Color temp attribute is not exposed while the lamp is off
-    assert light_state.attributes.get("color_temp_kelvin") is None
-
-    # test previous zero brightness received was ignored and brightness is restored
-    # see if the latest color_temp value received is restored
-    async_fire_mqtt_message(hass, "test_light_rgb", '{"state":"ON"}')
-    light_state = hass.states.get("light.test")
-    assert light_state.attributes.get("brightness") == 128
-    assert light_state.attributes.get("color_temp_kelvin") == 4975  # 201 mired
-
-    # A `0` brightness value is ignored when a light is turned on
-    async_fire_mqtt_message(hass, "test_light_rgb", '{"state":"ON","brightness":0}')
-    light_state = hass.states.get("light.test")
-    assert light_state.attributes.get("brightness") == 128
-
-
-@pytest.mark.parametrize(
-    "hass_config",
-    [
-        {
-            mqtt.DOMAIN: {
-                light.DOMAIN: {
-                    "schema": "json",
-                    "name": "test",
-                    "state_topic": "test_light_rgb",
-                    "command_topic": "test_light_rgb/set",
-                    "brightness": True,
-                    "color_temp": True,
                     "color_temp_kelvin": True,
                     "effect": True,
-                    "rgb": True,
-                    "xy": True,
-                    "hs": True,
+                    "supported_color_modes": ["color_temp", "hs"],
```

Comment:
Test was updated use `supported_color_modes` instead of the removed legacy `color_mode` support